book/managers_guide.md

```diff
@@ -0,0 +1,90 @@
+
+* Configuration of the code that might change should be easily identified from the code.
+* Ideally separate configuration files are used to store these options.
+
+### How are you storing input data?
+
+* Input data should be versioned, so that analysis outputs can be reproduced.
+* Data should be stored in an open format (e.g. CSV and ODS), not formats that are restricted to proprietary software like SAS and Stata. 
+* Large or complex datasets should be stored in a database.
+
+### How have you specified the analysis?
+
+* The code should have an accompanying specification.
+* The specification should show how data are transferred from function to function.
+* The specification helps reviewers to understand the code. Well-specified code is easier to understand and maintain.
+
+### What does the documentation look like?
+
+* Every function should be documented.
+* Function documentation should say what goes in and what comes out of each function.
+* Documentation should be easy to read, accessible, and stored closely to the code.
+* Good documentation includes usage examples and test data.
+
+### How has peer review been done?
+
+* Skilled colleagues should conduct internal peer reviews of the code.
+* If the product is high risk an external peer review should also be conducted.
+* Peer review should follow a standard procedure.
+
+### How have you kept track of who made changes and for which reasons?
+
+* The code, documentation, and peer review should all be version controlled. Git software is most commonly used for this.
+* There should be a clear record of every change and who made it.
+* Each change should have a reason.
+* Reviews should be stored with the version of the analysis that was reviewed.
+
+### How have you tested the code?
+
+* Testing means making sure that the code produces the right outputs for realistic example input data.
+* Good testing is automated 'unit' testing.
+* Each function and the end-to-end analysis should be tested using minimal, realistic data.
+
+### What have you tested?
+
+* Aim for every single function being tested
+* Tests should account for edge cases like missing data, zeroes, infinities, negative numbers, wrong data types, and invalid inputs.
+* Reviewers should sign-off that there is enough testing.
+
+### What are the dependencies?
+
+* Dependencies include anything the code needs to run, including software and package versions.
+* It can be difficult to run code on other systems when dependencies are not documented.
+* Documenting dependencies allows others to reproduce the analysis more easily.
+
+### Which other systems have you run this on?
+
+* Code should not be dependent on a specific computer to run. Running it on a colleague's system can help to check this. 
+* Container systems, like Docker, help to create reproducible environments to run code.
+### How would we do a dual run?
+
+* For the highest risk analysis, the program should be written twice separately or with different tools.
```

Comment:
I think we need to flesh this out more. Currently, it gives the impression that we recommend creating and maintaining indefinitely two separate pipelines for each high risk analysis, rather than as an early development/interim stage.

Reply:
I've tried to clarify this, but may still not be quite right?